tests/ui/default_iter_empty.rs

```diff
@@ -0,0 +1,10 @@
+// run-rustfix
+#![warn(clippy::default_iter_empty)]
+use std::collections::HashMap;
+
+fn main() {
+    // Do lint.
+    let _ = std::iter::Empty::<usize>::default();
+    let _ = std::iter::Empty::<HashMap<usize, usize>>::default();
+    let _foo: std::iter::Empty<usize> = std::iter::Empty::default();
```

Comment:
Could do with some tests with `use std::iter::Empty`, maybe some for `Default::default()` (probably shouldn't lint)

Reply:
I fixed in https://github.com/rust-lang/rust-clippy/pull/8989/commits/9fd048e744d57b338565defd235fbfd9f9d386fd